iothub_client/src/iothubtransport_amqp_device.c

```diff
@@ -564,6 +564,12 @@ static void internal_destroy_device(AMQP_DEVICE_INSTANCE* instance)
             twin_messenger_destroy(instance->twin_messenger_handle);
         }
 
+        // close the cbs before destroying the authentication handle 
+        if (instance->cbs_handle != NULL)
```

Comment:
This would break device multiplexing. the cbs_handle is a shared instance with different devices. The CBS handle should only be destroyed if calling amqp_connection_destroy.

the "amqp_connection" owns the cbs handle. And the AMQP transport owns the "amqp connection" and the "amqp device".
"amqp device" does not own the cbs_handle.

Reply:
There should be no memory leak if the proper destroy functions are called.

Reply:
ahh good to know